test/e2e/azure_test.go

```diff
@@ -267,6 +267,31 @@ var _ = Describe("Workload cluster creation", func() {
 		})
 	})
 
+	Context("Creating a Flatcar cluster [OPTIONAL]", func() {
+		It("With Flatcar control-plane and worker nodes", func() {
+			clusterName = getClusterName(clusterNamePrefix, "flatcar")
+
+			clusterctl.ApplyClusterTemplateAndWait(ctx, clusterctl.ApplyClusterTemplateAndWaitInput{
+				ClusterProxy: bootstrapClusterProxy,
+				ConfigCluster: clusterctl.ConfigClusterInput{
+					LogFolder:                filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName()),
+					ClusterctlConfigPath:     clusterctlConfigPath,
+					KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
+					InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+					Flavor:                   "flatcar",
```

Comment:
flavor needs to be defined in fake github release in azure-dev.yaml

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1729/files#diff-35f348a5d567ebeba39b1e40bbab0e1a7f64009cbadff142952f9b9691e94666R136

Reply:
Done IIUC.